### PR TITLE
SI-6040 error on unauthorized extension Dynamic

### PR DESCRIPTION
**Pull request by adriaanm**
authorization is easy to get: `import language.dynamics`

review by @hubertp

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -1586,12 +1592,18 @@ trait Typers extends Modes with Adaptations with Tags {
...
+
           if (parents exists (p => p != parent && p.tpe.typeSymbol == psym && !psym.isError))
             pending += ParentInheritedTwiceError(parent, psym)
+
+          validateDynamicParent(psym)
         }
+
+      if (!parents.isEmpty && parents.forall(!_.isErrorTyped)) {
+        val superclazz = parents.head.tpe.typeSymbol
+        validateDynamicParent(superclazz)
```
Comment (hubertp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: That looks like (unnecessary?) double checking. superclazz will be validated at the end of validateParentClass, right?

Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: d'oh! good catch

**Comment by adriaanm** [comment]
updated the commit based on Hubert's review

**Comment by adriaanm** [comment]
PLS REBUILD ALL

**Comment by hubertp** [comment]
lgtm

**Pull request merged**